File: src/tf/transform_broadcaster.cpp
#include "torch-tf.h"
#include <tf/transform_broadcaster.h>

TFIMP(tf::TransformBroadcaster *, TransformBroadcaster, new)()
{
  return new tf::TransformBroadcaster();
}

TFIMP(void, TransformBroadcaster, delete)(tf::TransformBroadcaster *self)
{
  delete self;
}

TFIMP(void, TransformBroadcaster, sendTransform)(tf::TransformBroadcaster *self, tf::StampedTransform *transform)
{
  self->sendTransform(*transform);
}

File: lua/tf/TransformBroadcaster.lua
local ffi = require 'ffi'
local torch = require 'torch'
local ros = require 'ros.env'
local utils = require 'ros.utils'
local tf = ros.tf

local TransformBroadcaster = torch.class('tf.TransformBroadcaster', tf)

function init()
  local TransformBroadcaster_method_names = {
    "new",
    "delete",
    "sendTransform"
  }
  
  return utils.create_method_table("tf_TransformBroadcaster_", TransformBroadcaster_method_names)
end

local f = init()

function TransformBroadcaster:__init()
  self.o = f.new()
end

function TransformBroadcaster:sendTransform(stampedTransform)
  f.sendTransform(self.o, stampedTransform:cdata())
end
